### ISSUE TITLE
Page editor: Double-click to edit template part

### ISSUE THREAD

**Issue by jameskoster** [WordPress/gutenberg#63305]
![double](https://github.com/WordPress/gutenberg/assets/846565/cd64467e-6546-407f-b24b-6dda77837292)

When page editing with the template visible, if you double-click on a template block a prompt appears directing you to the template editor. 

We could try doing something similar with template parts. To edit the header or footer currently you have to select the block, then click the 'Edit' link in the toolbar. In addition we could try listening for double-clicks, and send the user to edit the template part in focus mode.

**Comment by richtabor** [comment]
I think this is a great idea.

**Comment by noisysocks** [comment]
I constantly find myself double clicking on synced patterns thinking that it will do something 😅

**Comment by ramonjd** [comment]
Should all double clicks trigger edit? E.g., on a paragraph block inside a footer template part:

https://github.com/user-attachments/assets/5b13441f-34ca-48ab-8263-9a0a2d37c3fa

If "no", I'm wondering how far up the chain the double-click should propagate 🤔 maybe just the template part wrapper and its first inner block. 

A group block, for example, that wraps inner content in the footer (and has the same dimensions) will always capture the double click, but I suppose we could check the event target and dispatch a new Event to ensure the event is always treated as if it occurred on the parent template part. 

I dunno, I'm just rambling...

**Comment by jameskoster** [comment]
@ramonjd I think you're editing a template. In that scenario template part inner blocks can be directly manipulated. 

This issue is about the Page editor specifically, where template parts can be selected, but their child blocks are inaccessible until you click 'Edit' in the toolbar.

Personally I don't find the difference in behavior to be very helpful, but that's a separate issue 😪

**Comment by ramonjd** [comment]
> I think you're editing a template. In that scenario template part inner blocks can be directly manipulated.

> This issue is about the Page editor specifically, where template parts can be selected, but their child blocks are inaccessible until you click 'Edit' in the toolbar.

Oh, thanks for steering me back on course. I was blinded by excitement 😅 Maybe it's more straight forward than I thought then

**Comment by ramonjd** [comment]
Okay, update 😄 

I had a bit of time to try something:

- https://github.com/WordPress/gutenberg/pull/65024

_Question_: should the double click be restricted to the site editor? 

My PR enables it when editing pages in the site editor, but also when editing post/pages with the template shown in the post editor. Seemed logical to leave it that way to me.

Also I noticed the `EditTemplateBlocksNotification` double click event was still active in the background - I made it so it doesn't activate when a user clicks on a template part.
